src/components/Port/Port.wrapper.tsx

```diff
@@ -45,19 +45,15 @@ export class PortWrapper extends React.Component<IPortWrapperProps> {
 
   private nodeRef = React.createRef<HTMLDivElement>()
 
+  public componentDidMount () {
+    this.updatePortPosition()
+  }
+
   public componentDidUpdate (prevProps: IPortWrapperProps) {
-    // Update port position after a re-render if node.size has changed
-    if (!isEqual(this.props.node.size, prevProps.node.size)) {
-      const el = ReactDOM.findDOMNode(this.nodeRef.current) as HTMLInputElement
-      if (el) {
-        // Ports component should be positions absolute
-        // Factor this in so we get position relative to the node
-        const nodesEl = el.parentElement
-          ? el.parentElement
-          : { offsetLeft: 0, offsetTop: 0 }
-        // update port position after node size has been determined
-        this.props.onPortPositionChange({ config: this.props.config, node: this.props.node, port: this.props.port, el, nodesEl })
-      }
+    // Update port position after a re-render if there are more ports on the same side
+    // or if node.size has changed
+    if (this.portsOfType(this.props) !== this.portsOfType(prevProps) || !isEqual(this.props.node.size, prevProps.node.size)) {
```

Comment:
I've realised that `this.portsOfType(this.props) !== this.portsOfType(prevProps)` is only true when a new port is added to a node, not when existing ports shift position. 

When existing ports shift position, then `onComponentDidMount` happens for the port in its new position, but `onComponentDidMount` does not happen for the other ports on the same side that may have shifted position to accommodate the new port are.

A parent component (e.g. `<Ports />`) should pass a prop to each `<Port />` to indicate that they need to update their position in the chart state.